src/v/config/property.h

```diff
@@ -35,13 +35,18 @@ class property : public base_property {
       T def = T{},
       property::validator validator = property::noop_validator)
       : base_property(conf, name, desc, req)
-      , _value(std::move(def))
+      , _value(def)
```

Comment:
i know the commit says we `avhe to store the defaults ...` but why is that the case? why not return the mem state.

Reply:
In Kafka they always return default value even if it was overridden by value set in configuration file. f.e:
```
  log.retention.hours=1 sensitive=false synonyms={STATIC_BROKER_CONFIG:log.retention.hours=1, DEFAULT_CONFIG:log.retention.hours=168}
  ```

Reply:
I just don't think this makes sense, specially as we move to a unified configuration next month, where there is *exactly* __one__ configuration per cluster.

But I am OK with that until that PR for reconfiguration changes. This is a big area where Redpanda can make a big difference for system admins, and I think the Kafka API is too complex on this front. 

In any case. this is not a blocker for this PR. 

However please make sure you address this in the #553 and follow up PRs. 

related #241